### PR TITLE
Add hex.repo subtask to show repo config

### PR DESCRIPTION
**Pull request by parihar**
Fixes https://github.com/hexpm/hex/issues/363.

When repo is found:
```sh
$ mix hex.repo show hexpm
URL                  Public key                                          Auth key
https://repo.hex.pm  SHA256:O1LOYhHFW4kcrblKAxROaDEzLD8bn1seWbe5tq8TRsk
```

When repo is not found:
```sh
$ mix hex.repo show somerepo
** (Mix) Config does not contain repo somerepo
```

### CODE REVIEW

File: test/mix/tasks/hex.repo_test.exs [hunk 1]
```diff
@@ -48,4 +48,23 @@ defmodule Mix.Tasks.Hex.RepoTest do
       assert ["$repos": %{"reponame" => %{url: "other_url"}}] = Hex.Config.read
     end
   end
+
+  test "show returns repo config" do
+    in_tmp fn ->
+      Hex.State.put(:home, System.cwd!)
+
+      Mix.Tasks.Hex.Repo.run(["add", "reponame", "url"])
+      assert %{url: "url"} = Mix.Tasks.Hex.Repo.run(["show", "reponame"])
```
Comment (ericmj) [inline test/mix/tasks/hex.repo_test.exs]: We shouldn't return things from tasks. Technically tasks are specced to only return `:ok` so we need to test the output with `assert_received`.

**Comment by parihar** [comment]
I wasn't sure how to test with `assert_received`, so tried with `capture_io`.

**Comment by wojtekmach** [comment]
Please see eg
https://github.com/parihar/hex/blob/664b7a5947f495cdf8dd5c98032dbe068c566bab/test/mix/tasks/hex.info_test.exs
how to use assert_receive

W dniu śr., 10.05.2017 o 12:53 Abhi Parihar <notifications@github.com>
napisał(a):

**Comment by parihar** [comment]
`assert received` is definitely cleaner. Would you like these commits squashed?

**Comment by parihar** [comment]
Cool. Thanks for the help.

**Pull request merged**